### ISSUE TITLE
Objects appearing behind application

### ISSUE THREAD

**Issue by adgonz90** [blackberry/WebWorks-TabletOS#55]
Objects like Javascript Alerts and Confirms, HTML Selects, and the Camera UI are appearing behind the application.
We see a glimpse of the object when switching from the app to the main Playbook screen (where all the app icons are).
As of now, im not entirely sure of what is causing this problem. We are using several frameworks for our application.
RequireJS, jQuery, jQTouch, TouchSwipes, LawnChair, Phonegap (not loaded for Playbook).

This started occurring after updating to OS 2.0 (worked fine during last beta 7111).
I will keep attempting to find out the cause, but we are on a deadline and might have to create work arounds for these events on the Playbook. Unfortunately I can not give out any of the application's code but I will post the code for a test program that can recreate the problem, once I figure out what is causing it.

Thanks for your help,
Anthony Gonzalez

**Comment by erhorim** [comment]
Hi Tony, if you can produce a sample app that shows the error you saw, I'd be happy to investigate further for you.

Eric

**Comment by devbar** [comment]
I have the same issue with a self written FileDialog extension. I hope this will be fixed soon or there is a workaround. It's a really hard bug for my users :(

**Comment by erhorim** [comment]
Hey devbar, would you have a sample test app that shows this problem.
If so, feel free to share it so we may investigate further for you.

Thanks,
Eric

**Comment by devbar** [comment]
Hi Eric, 

I put it here http://dl.dropbox.com/u/2559251/IssueFocus.zip. It's as simple it can be. There is only a upload element. To reproduce just start the app and push the button. In my scenario it fails in 6 of 10 cases. 

BUT! I found an interesting fact while testing. If my application has no splash screen, I can't reproduce the issue. It works in 10 of 10 cases.

So this is my first workaround.

**Comment by adgonz90** [comment]
Hello All,

@erhorim, Im not sure if you ever received my reply. I replied to the Github email instead of posting here.

I tried devbar's solution of removing the splash screen and it seems like it is working properly now.
We are currently working with another RIM developer for the playbook and will let them know about this.

Thanks!

**Comment by erhorim** [comment]
@tony902304 , sorry I didn't check the Github email, I normally just checked the postings.
@devbar , thanks, we'll try out your sample app to investigate the problems.

Eric

**Comment by devbar** [comment]
Are there any news about this? 

best regards

**Comment by svzi** [comment]
What about this issue?

I really need a splash-screen for my app, because I use some JS frameworks that load really slow. Without the splash-screen my customers thinks the app is broken, because they only see the grey screen... But with the splash-screen enabled, I'm often not able to use some JS dialogs, because they appear to be behin the application.

Cheers,
Sven

**Comment by asiayeah** [comment]
I've exactly the same problem. I still keep my splash screen, but solved that by removing onFirstLaunch="true" from rim:loadingScreen. Could you give that a try?

Mine config is now something like:

```
  <rim:loadingScreen backgroundColor="#FFFFFF"
    foregroundImage="images/splash-lite.jpg">
  </rim:loadingScreen>
```

**Comment by svzi** [comment]
Thanks for your help, that works for me too. But I observed that the loading screen shows only for a second and than hides? So for the next 9 seconds the grey screen appears again and the app looks like dead... :-(

Any thoughts on this?

**Comment by asiayeah** [comment]
Yes, indeed, the splash screen appears and disappears quickly (~1 to 2 sec). Then it should proceed to display your app, which is under your control.

If your app needs 9 seconds to initialize, you could show a static image (e.g. the same splash screen or another fake image of your 1st screen) before initializing your app. In this way, the users won't be looking at a grey screen for 9 seconds.

**Comment by svzi** [comment]
Often the simplest ideas are the best. Did not know why I did not think on this... 

Thanks for your help asiayeah, I really appreciate it!

**Comment by srikanth4b9** [comment]
Hi,

I m developing playbook application using web works. The problem i m facing here "drop list popups" are showing behind the screen in the actual device..

I didn't see the problem in playbook simulator...

Here the snap shotlaybook popup issue

The snapshot Link: 
http://supportforums.blackberry.com/t5/Web-and-WebWorks-Development/Playbook-popup-are-coming-behind-the-screen/td-p/1885081

**Comment by devbar** [comment]
This is a bad bug because it's hard to bring the problems in application back to this issue. I hope it will be fixed soon :(

**Comment by srikanth4b9** [comment]
This is working fine when the splash screen not applied in the config.xml...

Now the problem is how to set the Splash screen... how to resolve both the issues...

**Comment by asiayeah** [comment]
Pardon me to repost:

I've exactly the same problem. I still keep my splash screen, but solved that by removing onFirstLaunch="true" from rim:loadingScreen. Could you give that a try?

Mine config is now something like:

```
  <rim:loadingScreen backgroundColor="#FFFFFF"
    foregroundImage="images/splash-lite.jpg">
  </rim:loadingScreen>
```

**Comment by srikanth4b9** [comment]
asiayeah : If your app needs 9 seconds to initialize, you could show a static image (e.g. the same splash screen or another fake image of your 1st screen) before initializing your app. In this way, the users won't be looking at a grey screen for 9 seconds.

How are you showing the static splash screen before initialising the the app...
I m applying the splash screen without the flag "onFirstLaunch" its showing only the 1 to 2 seconds. and the white color  background showing...In my idex file i'm adding one more image its showing after the 10 seconds..So how to set image in the white color background time (Means : 2 sec to 10 sec duration time)..
